crates/libs/windows/src/core/agile_reference.rs

```diff
@@ -0,0 +1,30 @@
+use super::*;
+use bindings::*;
+use core::marker::PhantomData;
+
+/// A safe wrapper around WinRT interfaces.
+///
+/// Some interfaces are not marked as agile thus then don't implement [`Send`].
+/// These interfaces can be made agile through this `AgileReference`.
+pub struct AgileReference<T> {
+    reference: IAgileReference,
+    _marker: PhantomData<T>,
+}
+
+impl<T: Interface> AgileReference<T> {
+    /// Creates a new wrapper around the reference
+    pub fn new<'a>(from_ref: &'a T) -> Result<Self>
+    where
+        &'a T: IntoParam<'a, IUnknown>,
```

Comment:
```suggestion
    pub fn new<'a, T: IntoParam<'a, IUnknown>>(object: T) -> Result<Self> {
```

Reply:
This breaks type inference because Rust can't tell that it will return a `Self<T>` but I don't see any way around it.

Reply:
Hmm - not having the type inferred seems like a recipe for pain and suffering. The test should validate that this (somehow) works.